articles/quickstart/native/ios-objc/03-user-sessions.md

```diff
@@ -211,15 +209,17 @@ A0SimpleKeychain *keychain = [[A0SimpleKeychain alloc] initWithService:@"Auth0"]
 [keychain clearAll];
 ```
 
-### Optional: Encapsulate session handling
+## Optional: Encapsulate Session Handling
+
+Handling user sessions is not a straightforward task. You can simplify it by storing token-related information and processes in a class. The class separates the logic for handling user sessions from the View Controller layer. 
 
-As you have probably realized by now, session handling is not a straightforward process. All this token-related information and processes can be encapsulated into a class that separates its logic from the View Controller layer. We recommend that you download the sample project from this tutorial and take a look at its implementation, focusing on the `SessionManager` class, which is in charge of dealing with these processes.
+We recommend that you download the sample project from this tutorial and look at its implementation. Focus on the `SessionManager` class, which manages the session handling processes.
 
-## Fetch the User Profile
+## Get the User Profile
 
-The first step is to fetch the user profile. To do so, you need a valid `accessToken` first.
+To get the user profile, you need a valid access token. 
 
-You need to call a method from the `Auth0` module that allows you to fetch the user profile given an `accessToken`:
+From the `Auth0` module, call a method that allows you to get the user profile:
```

Comment:
This line seems redundant, could just go straight to snippet.

Reply:
We generally try not to leave code snippets without introduction. The general idea is to introduce code snippets with a brief explanation of their purpose.

Reply:
ok fine with having an intro just feel the text sounds odd `call a method`, be better if you said what method you will call.

Reply:
I agree, do you know which method should be called? Is it `userInfoWithAccessToken`?

Reply:
Yes https://github.com/auth0/Auth0.swift/blob/master/Auth0/Authentication.swift#L342